### PR TITLE
Combine feature request mode samples

### PR DESCRIPTION
**Pull request by Rachael-E**
This PR combines the three existing samples that showcase our different service feature table request modes into one. 

- Deletes Service Feature Table Cache and Service Feature Table No Cache samples
- Replaces Service Feature Table Manual Cache with a new sample Toggle Between Feature Request Modes -> since this is a new sample I have also updated the copyright on the file.
- New sample has new data, and a new UI (three radio buttons to flip between different request modes, and a populate button to carry out the manual cache mode).

@jenmerritt could you review please?

### CODE REVIEW

File: feature_layers/toggle-between-feature-request-modes/README.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+## How to use the sample
+
+Choose a request mode by clicking on a radio button. Pan and zoom to see how the features update at different scales. If you choose "Manual Cache", click the "Populate" button to manually get a cache with a subset of features.
+
+Note: The service limits requests to 2000 features.
+
+## How it works
+
+1. Create an `ServiceFeatureTable` with a feature service URL.
```
Comment (jenmerritt) [inline feature_layers/toggle-between-feature-request-modes/README.md]:
```suggestion
1. Create a `ServiceFeatureTable` with a feature service URL.
```

File: feature_layers/toggle-between-feature-request-modes/README.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+2. Set the `FeatureRequestMode` property of the service feature table to the desired mode (`ON_INTERACTION_CACHE`, `ON_INTERACTION_NO_CACHE`, or `MANUAL_CACHE`).
+    * If using `MANUAL_CACHE`, populate the features with `ServiceFeatureTable.populateFromServiceAsync()`.
+3. Create a `FeatureLayer` with the feature table and add it to an `ArcGISMap's` operational layers to display it.
```
Comment (jenmerritt) [inline feature_layers/toggle-between-feature-request-modes/README.md]:
```suggestion
3. Create a `FeatureLayer` with the feature table and add it to an `ArcGISMap`'s operational layers to display it.
```

File: feature_layers/toggle-between-feature-request-modes/src/main/java/com/esri/samples/toggle_between_feature_request_modes/ToggleBetweenFeatureRequestModesSample.java [hunk 3]
```diff
@@ -0,0 +1,242 @@
...
+      // create a feature layer from the service feature table
+      featureLayer = new FeatureLayer(featureTable);
+      // set up the control panel for switching between request modes
+      setUpUi();
+
+      // listen for when a radio button within the toggle group is selected
+      toggleGroup.selectedToggleProperty().addListener(e -> {
+
+        // check if the feature layer has already been added to the map's operational layers, and if not, add it
+        if (map.getOperationalLayers().size() == 0){
```
Comment (jenmerritt) [inline feature_layers/toggle-between-feature-request-modes/src/main/java/com/esri/samples/toggle_between_feature_request_modes/ToggleBetweenFeatureRequestModesSample.java:100]:
```suggestion
        if (map.getOperationalLayers().isEmpty()){
```

File: feature_layers/toggle-between-feature-request-modes/src/main/java/com/esri/samples/toggle_between_feature_request_modes/ToggleBetweenFeatureRequestModesSample.java [hunk 4]
```diff
@@ -0,0 +1,242 @@
...
+    // get queried features from service feature table and clear previous cache
+    ListenableFuture<FeatureQueryResult> tableResult = featureTable.populateFromServiceAsync(queryParams, true, outfields);
+
+    tableResult.addDoneListener(() -> {
+      try {
+        // find the number of features returned from query
+        AtomicInteger featuresReturned = new AtomicInteger();
+        tableResult.get().forEach(feature -> featuresReturned.getAndIncrement());
+
+        // display to user how many features where returned
```
Comment (jenmerritt) [inline feature_layers/toggle-between-feature-request-modes/src/main/java/com/esri/samples/toggle_between_feature_request_modes/ToggleBetweenFeatureRequestModesSample.java]:
```suggestion
        // display number of returned features to the user
```

File: feature_layers/toggle-between-feature-request-modes/src/main/java/com/esri/samples/toggle_between_feature_request_modes/ToggleBetweenFeatureRequestModesSample.java [hunk 5]
```diff
@@ -0,0 +1,242 @@
...
+        label.setText("Populated " + featuresReturned + " features.");
+      } catch (Exception e) {
+        // on any error, display the stack trace
+        e.printStackTrace();
+      }
+    });
+  }
+
+  /**
+   * Sets up control panel with radio buttons to toggle between feature request modes, a label displaying the features
```
Comment (jenmerritt) [inline feature_layers/toggle-between-feature-request-modes/src/main/java/com/esri/samples/toggle_between_feature_request_modes/ToggleBetweenFeatureRequestModesSample.java]:
```suggestion
   * Sets up a control panel with radio buttons to toggle between feature request modes, a label displaying the features
```

File: feature_layers/toggle-between-feature-request-modes/src/main/java/com/esri/samples/toggle_between_feature_request_modes/ToggleBetweenFeatureRequestModesSample.java [hunk 6]
```diff
@@ -0,0 +1,242 @@
...
+
+    // create a control panel
+    controlsVBox = new VBox(6);
+    controlsVBox.setBackground(new Background(new BackgroundFill(Paint.valueOf("rgba(0,0,0,0.5)"), CornerRadii.EMPTY,
+      Insets.EMPTY)));
+    controlsVBox.setPadding(new Insets(10.0));
+    controlsVBox.setMaxSize(200, 80);
+    controlsVBox.getStyleClass().add("panel-region");
+
+    // create a label to display number of features returned
```
Comment (jenmerritt) [inline feature_layers/toggle-between-feature-request-modes/src/main/java/com/esri/samples/toggle_between_feature_request_modes/ToggleBetweenFeatureRequestModesSample.java]:
```suggestion
    // create a label to display information on the UI
```
It has multiple purposes

**Review by jenmerritt — COMMENTED** [review]
@Rachael-E I've added a few tidy up comments. 

Also, I think there is maybe another step needed when selecting the request mode. If I quickly select a different mode the UI sometimes locks up, particularly if trying to switch over the Manual. Is Manual meant to be clear of features at the start each time? As sometimes it is and others if I click quickly there are features showing, so not sure if there's a timing issue at play. Could try disabling the UI to the user until the selected mode has finished loading just to guard against this?

**Comment by Rachael-E** [comment]
thanks @jenmerritt , I've set the vbox to disabled whilst the mapview is still drawing: this should guard against the behaviour you found. could you take another look please? thanks!

**Comment by jenmerritt** [comment]
@Rachael-E that works well when toggling. Only thing is the vbox and progress indicator seem to flicker between enabled/disabled at the start, perhaps as multiple things are initially drawing? Can send a gif / show you if it's my machine.

**Review by jenmerritt — APPROVED** [review]
@Rachael-E As per discussion about loading, approving as better to be definitely displaying the correct feature request modes which it now does.

**Pull request merged**